### DISCUSSION TITLE
SWA CLI vs Azure CLI

### DISCUSSION THREAD

**Discussion by wi5nia** [Azure/static-web-apps-cli#516] [General]
hi,

can someone elaborate why use this CLI instead of it being built into the Azure CLI? Just besides the argument that not everyone wants to install the Azure CLI but that doesn't make sense to me as if you are using the Azure Static Web Apps why not install the Azure CLI anyway.

**Comment by sgollapudi77** [comment]
Hey @wi5nia , with SWA_CLI you can have 

1. e2e testing experience locally using `swa start`
2. You can deploy directly to Azure using `swa deploy`. Of course you can do this using Azure CLI also but it maynot be as seamless as SWA_CLI because SWA_CLI also has custom [config support](https://azure.github.io/static-web-apps-cli/docs/cli/swa-config) thus eliminating the need of flags. 
3. Now with the new SWA_CLI version, we're supporting logging to azure also using `swa login`.

**Comment by manekinekko** [comment]
@wi5nia In addition to what @sgollapudi77 mentioned, SWA CLI provides a complete emulation of Azure Static Web Apps in your local dev environment. It also emulates local Auth, and supports `staticwebapp.config.json` rules locally, etc. 

Here is a short comparison table, in case this can help:

|                                        | Azure CLI | SWA CLI |
| -------------------------------------- | --------- | ------- |
| Login to Azure                         | ✅        | ✅      |
| Provision SWA instance                     | ✅        | ✅      |
| Deploy SWA to Azure                    | ✅        | ✅      |
| Azure Function Core tools integration  |           | ✅      |
| Reading local GitHub workflow          |           | ✅      |
| Reading `staticwebapp.config.json`     |           | ✅      |
| Serving the application static content |           | ✅      |
| Proxying API requests|           | ✅      |
| Emulating SWA Authentication           |           | ✅      |
| Emulating SWA Authorizations           |           | ✅      |
| Initializing new SWA project           |           | ✅      |
| Framework detection capabilities       |           | ✅      |

**Comment by wi5nia** [comment]
Thanks for the explanation. But my question also was why not include all of this in Azure CLI?
